### PR TITLE
Add "ast" locale support

### PR DESCRIPTION
**Pull request by softastur**
Asturian is a language spoken in the north-east of Iberian Peninsula by 500000 speakers, and it belongs to the Astur-leonese linguistic branch. This PR adds support for "ast" locale. Translations are 20% completed.

However, we should keep an eye on issue https://github.com/bluesky-social/social-app/issues/6346 before merging

### CODE REVIEW

**Comment by iyanuvieu** [comment]
Yes, please! Let us, asturian language community to spread in BS! 😇

**Comment by pfrazee** [comment]
Great! TYSM! Is there a native speaker available who could review this?

**Comment by softastur** [comment]
> Great! TYSM! Is there a native speaker available who could review this?

Softastur is a non-profit organization which localizes free (as freedom) software into Asturian and it is ran by volunteers. We started our activity in 2003, and Firefox and AOSP are our current biggest projects. We also have our glossary and make use of Academia de la Llingua Asturiana (Academy for Asturian Language) grammar, the only standard for Asturian language, created by Asturian Government in 1980. More info [here](https://en.wikipedia.org/wiki/Academy_of_the_Asturian_Language) and [here](https://ast.wikipedia.org/wiki/Softastur) (in Asturian)

File: src/locale/languages.ts [hunk 1]
```diff
@@ -42,6 +43,7 @@ interface AppLanguageConfig {
 export const APP_LANGUAGES: AppLanguageConfig[] = [
   {code2: AppLanguage.en, name: 'English'},
   {code2: AppLanguage.an, name: 'Aragonés – Aragonese'},
+  {code3: AppLanguage.ast, name: 'Asturianu - Asturian'},
```
Comment (gaearon) [inline src/locale/languages.ts:46]: How does this work? I don't think the types match up (currently `code2` is the required field on this type, and all the application logic also checks for `code2` here).

I think this needs a change in the application code to work.

**Pull request merged**